### PR TITLE
fix: `--version` for amplify and disable subcommands version

### PR DESCRIPTION
**Pull request by 0618**
*Issue #, if available:*

*Description of changes:*

**Issue**: `amplify --version` and `amplify <subcommand> --version` don't return correct versions. Always return `1.0.0`.
**Reason**: It's because `yargs` always gets the package.json version from the root dir, which means it gets custemer's project version.
**Changes in this PR**:
- Disable subcommand's version
- Use absolute path to get the version from `backend-cli` package
- Add test

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/cli/src/main_parser_factory.ts [hunk 1]
```diff
@@ -8,7 +10,12 @@ import { createConfigureCommand } from './commands/configure/configure_command_f
  * Creates main parser.
  */
 export const createMainParser = (): Argv => {
+  const absolutePath = './node_modules/@aws-amplify/backend-cli';
```
Comment (sobolk) [inline packages/cli/src/main_parser_factory.ts:13]: we should use `import.meta.url` to determine this.
But I suggest to wait until @Amplifiyer 's PR goes in that adds components for that to platform core.

File: .changeset/wet-items-raise.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/backend-cli': patch
+---
+
+`amplify sandbox --version` returns version of the sandbox package
```
Comment (edwardfoyle) [inline .changeset/wet-items-raise.md:5]: Seems like this message needs to be updated

File: packages/cli/src/main_parser_factory.test.ts [hunk 3]
```diff
@@ -16,6 +17,11 @@ void describe('main parser', { concurrency: false }, () => {
     assert.match(output, /generate\s+Generates post deployment artifacts/);
   });
 
+  void it('shows version', async () => {
+    const output = await commandRunner.runCommand('--version');
+    assert.match(output, RegExp(version));
```
Comment (edwardfoyle) [inline packages/cli/src/main_parser_factory.test.ts:22]: Can this just be `assert.equal(output, version)`?

File: packages/cli/src/commands/configure/configure_command.ts [hunk 4]
```diff
@@ -36,6 +36,7 @@ export class ConfigureCommand implements CommandModule<object> {
    */
   builder = (yargs: Argv): Argv => {
     return yargs
+      .version(false)
```
Comment (edwardfoyle) [inline packages/cli/src/commands/configure/configure_command.ts:39]: Is there any way we can set this globally on the yargs parser?

File: .changeset/wet-items-raise.md (hunk 2 above)
Comment (0618) [reply .changeset/wet-items-raise.md:5]: Good catch!

File: packages/cli/src/main_parser_factory.ts (hunk 1 above)
Comment (Amplifiyer) [reply packages/cli/src/main_parser_factory.ts:13]: You can now use 

https://github.com/aws-amplify/amplify-backend/blob/70685f36bdd089711fab3ba6eab0c375dad35827/packages/platform-core/src/package_json_reader.ts#L8

See example 

https://github.com/aws-amplify/amplify-backend/blob/70685f36bdd089711fab3ba6eab0c375dad35827/packages/cli/src/commands/sandbox/sandbox_command_factory.ts#L57-L59

File: packages/cli/src/commands/configure/configure_command.ts (hunk 4 above)
Comment (Amplifiyer) [reply packages/cli/src/commands/configure/configure_command.ts:39]: We decided it's ok to let the `--version` works for every command right? so just setting on the main_parser_factory is sufficient?

Comment (0618) [reply packages/cli/src/commands/configure/configure_command.ts:39]: We mentioned about having them returning the same version or disable the versions for sub-commands, but didn't have a hard conclusion.
IMO, having `amplify sandbox --version` return the same version as `amplify --version` might be confusing. It makes more sense to just disable `--version` for the sub-commands.

Comment (sobolk) [reply packages/cli/src/commands/configure/configure_command.ts:39]: I think `--version` doesn't make much sense on subcommands, but I don't think it matters much, so I'm fine either way, with preference of disabling this on subcommands.

File: packages/cli/src/main_parser_factory.test.ts (hunk 3 above)
Comment (0618) [reply packages/cli/src/main_parser_factory.test.ts:22]: Unfortunately, no. It'd have the following error

image

Comment (0618) [reply packages/cli/src/main_parser_factory.test.ts]: Done 33af02788

File: .changeset/wet-items-raise.md (hunk 2 above)
Comment (0618) [reply .changeset/wet-items-raise.md]: Done 24038c422

File: packages/cli/src/commands/configure/configure_command.ts (hunk 4 above)
Comment (edwardfoyle) [reply packages/cli/src/commands/configure/configure_command.ts:39]: My preference is also disabling on subcommands. It's just noise in the help printouts

File: packages/cli/src/main_parser_factory.ts (hunk 1 above)
Comment (0618) [reply packages/cli/src/main_parser_factory.ts]: Done! 4c50d21d4

**Review by sobolk — DISMISSED** [review]

File: packages/cli/src/commands/configure/configure_command.ts (hunk 4 above)
Comment (edwardfoyle) [reply packages/cli/src/commands/configure/configure_command.ts:39]: I'm assuming the verdict on this is that it has to be set on each subcommand?

**Review by edwardfoyle — APPROVED** [review]

File: packages/cli/src/commands/configure/configure_command.ts (hunk 4 above)
Comment (Amplifiyer) [reply packages/cli/src/commands/configure/configure_command.ts:39]: If we want to hide it from subcommands then I believe yes.

Comment (0618) [reply packages/cli/src/commands/configure/configure_command.ts:39]: Yea, the `--version` option is hidden from `--help` now

Comment (0618) [reply packages/cli/src/commands/configure/configure_command.ts:39]: > Is there any way we can set this globally on the yargs parser?

Unfortunately, I was not able to find a way to set it globally from yargs. If we want, I guess we can customize a `amplifyYargs`. What do you think? @edwardfoyle

Comment (edwardfoyle) [reply packages/cli/src/commands/configure/configure_command.ts:39]: Eh, probably not worth it at this point

**Review by sobolk — APPROVED** [review]

**Review by Amplifiyer — APPROVED** [review]

**Pull request merged**